lib/Habrok.js

```diff
@@ -78,13 +78,17 @@ Habrok.prototype.request = function request(raw, opts) {
     } else if (errorCode === ECONNRESET || statusCode > 299) {
       if ((!err && self.RETRY_CODES.indexOf(statusCode) === -1) || attempt >= self.RETRIES) {
         debug('fail', req);
-
+        if (has(options, 'debugRequest')) options.debugRequest({ attempt });
         return reject(err || boom.create(statusCode, null, body));
       }
 
       return setTimeout(
         () => {
           debug('retry', raw);
+          if (has(options, 'onRetriableFailure')) {
+            const returnedError = err || boom.create(statusCode, null, body);
+            options.onRetriableFailure(returnedError);
```

Comment:
Hmm, what do you think about just calling it `onRetry` the failure isn't really adding much info since we already know it only retries on failures.

The most important piece of information we want here is the status code. We don't necessarily want to pull that out every time, it would be nice to just have it.

`onRetry(statusCode, error)`

What do you think about that signature?